emerald/src/world/physics/physics_engine.rs

```diff
@@ -205,6 +206,48 @@ impl PhysicsEngine {
         Vec::new()
     }
 
+    #[inline]
+    pub fn intersections_with_ray(
+        &self,
+        ray_cast_query: RayCastQuery<'_>,
+    ) -> Vec<RayIntersectionResult> {
+        let ray_intersections = Arc::new(Mutex::new(Vec::new()));
```

Comment:
I hate using mutexes and unwrapping internally, but I'm not sure how else to implement this in this case.